treeherder/extract/extract_alerts.py

```diff
@@ -18,21 +16,27 @@
                             parse)
 from redis import Redis
 
+from treeherder.config.settings import REDIS_URL
 from treeherder.perf.models import PerformanceAlertSummary
 
 CONFIG_FILE = (File.new_instance(__file__).parent / "extract_alerts.json").abspath
 
-logger = logging.getLogger(__name__)
-logger.setLevel(logging.INFO)
-
 
 class ExtractAlerts:
     def run(self, force=False, restart=False, merge=False):
-        # SETUP LOGGING
-        settings = startup.read_settings(filename=CONFIG_FILE)
-        constants.set(settings.constants)
-        Log.start(settings.debug)
+        try:
+            # SETUP LOGGING
+            settings = startup.read_settings(filename=CONFIG_FILE)
+            constants.set(settings.constants)
+            Log.start(settings.debug)
+
+            self.extract(settings, force, restart, merge)
+        except Exception as e:
+            Log.error("could not extract alerts", cause=e)
+        finally:
+            Log.stop()
```

Comment:
@armenzg This is just wrapping in more try catch so I got better error reports from Heroku

Reply:
What do `start()` and `stop()` do?
Is it to enable debug logging only for a brief amount of time?

Reply:
@armenzg yes, the logging is shunted to `logging.getLogger(settings.debug.name)` for the duration of the extract.  Specifically, the `Log.stop()` is waiting for other threads' logs to drain before continuing.